File: find_duplicates.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('src/data/questions.json', 'utf8'));

let count = 0;
data.forEach(q => {
    if (q.text.it === q.text.en && q.text.en.length > 20) {
        console.log(`Question ID: ${q.id}`);
        console.log(`English: ${q.text.en}`);
        console.log(`Italian: ${q.text.it}`);
        console.log("--------------------");
        count++;
    }
    
    q.options.forEach(opt => {
        if (opt.text.it === opt.text.en && opt.text.en.length > 20) {
             const ignoreTerms = ["WPA3", "AES", "FIM", "SIEM", "IDS", "IPS", "OAuth", "Common Vulnerabilities and Exposures"];
             const isTechnical = ignoreTerms.some(term => opt.text.en.includes(term));
             if (!isTechnical) {
                console.log(`Option ID: ${opt.id} in ${q.id}`);
                console.log(`English: ${opt.text.en}`);
                console.log(`Italian: ${opt.text.it}`);
                console.log("--------------------");
                count++;
             }
        }
    });
});

console.log(`Found ${count} suspicious matches.`);

File: find_duplicates.py
import json

with open('src/data/questions.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

count = 0
for q in data:
    if q['text'].get('it') == q['text'].get('en') and len(q['text'].get('en', '')) > 20:
        print(f"Question ID: {q['id']}")
        print(f"English: {q['text']['en']}")
        print(f"Italian: {q['text']['it']}")
        print("-" * 20)
        count += 1
    
    for opt in q['options']:
        if opt['text'].get('it') == opt['text'].get('en') and len(opt['text'].get('en', '')) > 20:
             # Some options might be legitimate to be same (technical terms)
             # but long sentences usually shouldn't be.
             is_technical = any(term in opt['text']['en'] for term in ["WPA3", "AES", "FIM", "SIEM", "IDS", "IPS"])
             if not is_technical:
                print(f"Option ID: {opt['id']} in {q['id']}")
                print(f"English: {opt['text']['en']}")
                print(f"Italian: {opt['text']['it']}")
                print("-" * 20)
                count += 1

print(f"Found {count} suspicious matches.")
